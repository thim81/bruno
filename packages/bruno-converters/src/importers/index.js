export { importCollection as importPostmanCollection } from './postman-collection.js';
export { importEnvironment as importPostmanEnvironment } from './postman-environment.js';
export { importCollection as importInsomniaCollection } from './insomnia-collection.js';
export { importCollection as importOpenAPICollection } from './openapi-collection.js';

export { exportCollection as exportBrunoCollection } from './collections/export';
