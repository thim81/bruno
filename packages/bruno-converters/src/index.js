export { importCollection as importPostmanCollection } from './importers/postman-collection.js';
export { importEnvironment as importPostmanEnvironment } from './importers/postman-environment.js';
export { importCollection as importInsomniaCollection } from './importers/insomnia-collection.js';
export { importCollection as importOpenAPICollection } from './importers/openapi-collection.js';

export { exportCollection as exportBrunoCollection } from './collections/export';
