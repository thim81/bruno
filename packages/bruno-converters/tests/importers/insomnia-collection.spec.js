import { describe, it, expect } from '@jest/globals';
import path from 'path';
import { importInsomniaCollection } from '../../src';

describe('insomnia-collection', () => {
  it('should correctly import a valid Insomnia collection file', async () => {
    // Path to the sample Insomnia file
    const fileName = path.resolve(__dirname, '../data', 'sample_insomnia_export.json');

    // Call the importCollection function with the sample file
    const brunoCollection = await importInsomniaCollection(fileName);

    // Assert that the returned collection matches the expected structure
    expect(brunoCollection).toMatchSnapshot()
  });
});
