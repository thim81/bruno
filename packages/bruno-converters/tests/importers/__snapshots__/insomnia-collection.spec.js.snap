// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insomnia-collection should correctly import a valid Insomnia collection file 1`] = `
{
  "environments": [],
  "items": [
    {
      "items": [
        {
          "name": "Request1",
          "request": {
            "auth": {
              "basic": null,
              "bearer": null,
              "digest": null,
              "mode": "none",
            },
            "body": {
              "formUrlEncoded": [],
              "json": null,
              "mode": "none",
              "multipartForm": [],
              "text": null,
              "xml": null,
            },
            "headers": [],
            "method": "GET",
            "params": [],
            "url": "https://httpbin.org/get",
          },
          "seq": 1,
          "type": "http-request",
          "uid": "mockeduuidvalue123456",
        },
        {
          "name": "Request1",
          "request": {
            "auth": {
              "basic": null,
              "bearer": null,
              "digest": null,
              "mode": "none",
            },
            "body": {
              "formUrlEncoded": [],
              "json": null,
              "mode": "none",
              "multipartForm": [],
              "text": null,
              "xml": null,
            },
            "headers": [],
            "method": "GET",
            "params": [],
            "url": "https://httpbin.org/get",
          },
          "seq": 2,
          "type": "http-request",
          "uid": "mockeduuidvalue123456",
        },
      ],
      "name": "Folder1",
      "type": "folder",
      "uid": "mockeduuidvalue123456",
    },
    {
      "items": [
        {
          "name": "Request2",
          "request": {
            "auth": {
              "basic": null,
              "bearer": null,
              "digest": null,
              "mode": "none",
            },
            "body": {
              "formUrlEncoded": [],
              "json": null,
              "mode": "none",
              "multipartForm": [],
              "text": null,
              "xml": null,
            },
            "headers": [],
            "method": "GET",
            "params": [],
            "url": "https://httpbin.org/get",
          },
          "seq": 1,
          "type": "http-request",
          "uid": "mockeduuidvalue123456",
        },
        {
          "name": "Request2",
          "request": {
            "auth": {
              "basic": null,
              "bearer": null,
              "digest": null,
              "mode": "none",
            },
            "body": {
              "formUrlEncoded": [],
              "json": null,
              "mode": "none",
              "multipartForm": [],
              "text": null,
              "xml": null,
            },
            "headers": [],
            "method": "GET",
            "params": [],
            "url": "https://httpbin.org/get",
          },
          "seq": 2,
          "type": "http-request",
          "uid": "mockeduuidvalue123456",
        },
      ],
      "name": "Folder2",
      "type": "folder",
      "uid": "mockeduuidvalue123456",
    },
  ],
  "name": "Hello World Workspace Insomnia",
  "uid": "mockeduuidvalue123456",
  "version": "1",
}
`;
